src/main/java/parser/Parser.java

```diff
@@ -74,7 +76,21 @@ static Node parseNode(String concrete) {
                 } while (!concreteElements[i].contains("}"));
             }
             if (concreteElement.length() != 0) {
-                lst.add(function.apply(concreteElement));
+                // added : if returnValue of function is array, do addAll(), else do add()
+                T thing = function.apply(concreteElement);
+                if (thing instanceof List) {
```

Comment:
You should just make the `parseEdge` method always return a list.

Reply:
yes but in parseList's signature
    private static <T> List<T> parseList(String concrete, Function<String, T> function)
 Function<String, T>'s output type has to be T. It can't be changed to List because parseNode return a node.
if i make it return as a singletonlist and i still need to cast it to T.

Reply:
easiest solution is to just make `parseNode` return a list as well.

Reply:
yes, we don't necessary need to change, right? because T can be anything, it can be a list or an object.

Reply:
I don't like having the parseEdge method have two different return types, it's really weird to me.